api/views/curriculum.py

```diff
@@ -0,0 +1,145 @@
+    name = "curriculum"
+
+    def get_queryset(self):
+        """Return the list of experiences of the person."""
+        return Experience.objects.filter(curriculum_year__person=self.get_object())\
+            .order_by('curriculum_year__academic__year')
+
+    def get_experience(self) -> Optional[Experience]:
+        """Get the current experience from the uuid."""
+        return get_object_or_404(self.get_queryset(), pk=self.kwargs.get('xp'))
+
+
+class CurriculumExperienceListAndCreateView(CurriculumExperienceView):
+    def get(self, request, *args, **kwargs):
+        """Return the list of experiences from the person's CV."""
+        serializer = serializers.ExperienceOutputSerializer(self.get_queryset(), many=True)
+        return Response(serializer.data)
+
+    def post(self, request, *args, **kwargs):
+        """Add an experience to the person's CV."""
+        serializer = serializers.ExperienceInputSerializer(data=request.data, related_person=self.get_object())
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+
+        output_data = serializers.ExperienceOutputSerializer(serializer.instance).data
+        return Response(output_data, status=status.HTTP_201_CREATED)
+
+
+class CurriculumExperienceDetailUpdateAndDeleteView(CurriculumExperienceView):
+    def get(self, request, *args, **kwargs):
+        """Return a specific experience from the person's CV."""
+        serializer = serializers.ExperienceOutputSerializer(self.get_experience())
+        return Response(serializer.data)
+
+    def put(self, request, *args, **kwargs):
+        """Update one of the experiences from the person's CV."""
+        experience_to_update = self.get_experience()
+
+        if experience_to_update.doctorateadmission_set.exists():
+            raise PermissionDenied(_('This experience cannot be updated as it has already been valuated.'))
+
+        serializer = serializers.ExperienceInputSerializer(
+            instance=experience_to_update,
+            data=request.data,
+            related_person=self.get_object(),
+        )
+
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+
+        output_data = serializers.ExperienceOutputSerializer(serializer.instance).data
+        return Response(output_data)
+
+    def delete(self, request, *args, **kwargs):
+        """Remove one of the experiences from the person's CV."""
+        experience_to_delete = self.get_experience()
+
+        if experience_to_delete.doctorateadmission_set.exists():
+            raise PermissionDenied(_('This experience cannot be deleted as it has already been valuated.'))
+
+        experience_to_delete.delete()
+
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class CurriculumFileView(PersonRelatedMixin, APIPermissionRequiredMixin, RetrieveModelMixin, UpdateModelMixin,
+                         GenericAPIView):
```

Comment:
```suggestion
class CurriculumFileView(PersonRelatedMixin, APIPermissionRequiredMixin, UpdateModelMixin, RetrieveAPIView):
```
et retirer `def get()`